hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java

```diff
@@ -79,25 +79,29 @@ public SyntheticTxnFactory() {
 			final List<TokenTransferWrapper> tokenTransferWrappers
 	) {
 		final var builder = CryptoTransferTransactionBody.newBuilder();
-
+		boolean hasTokenTransfers = false;
 		for (final TokenTransferWrapper tokenTransferWrapper : tokenTransferWrappers) {
+			/*- changes inside a tokenTransferWrapper are always related to the same token -*/
 			for (final var nftExchange : tokenTransferWrapper.nftExchanges()) {
 				builder.addTokenTransfers(TokenTransferList.newBuilder()
```

Comment:
It doesn't make sense to add multiple `tokenTransferList`s here, because (as per your comment), we already know that all have the same `TokenID`.

We only need to add **one** `TokenTransferList` per `TokenID`; and we can include all the NFT exchanges in that one list.

It also follows there's no reason to change `PureTransferSemanticChecks`.